tests/Support/SupportConditionableTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Illuminate\Tests\Support;
+
+use Illuminate\Support\Traits\Conditionable;
+use PHPUnit\Framework\TestCase;
+
+class SupportConditionableTest extends TestCase
+{
+    public function testWhenConditionCallback()
+    {
+        $conditionTriggered = false;
+        $defaultTriggered = false;
+
+        $object = (new CustomConditionableObject())
+            ->when(2, function ($object, $condition) use (&$conditionTriggered) {
+                $conditionTriggered = true;
+                $object->on();
+                $this->assertEquals(2, $condition);
+            }, function ($object) use (&$defaultTriggered) {
+                $defaultTriggered = true;
+                $object->off();
+            });
+
+        $this->assertTrue($object->enabled);
+        $this->assertTrue($conditionTriggered);
+        $this->assertFalse($defaultTriggered);
+    }
```

Comment:
Why are all of these local by-reference variables needed? Is it not enough to simply inspect the object?

All of these tests could be vastly simplified, using an object like this:

```php
class ConditionableLogger
{
    use Conditionable;

    public $values = [];

    public function log(...$values)
    {
        $values[] = $values;
    }
}
```

Then, for example, the test above could become:

```php
$logger = (new ConditionableLogger())
    ->when(2, function ($logger, $condition) {
        $logger->log('triggered', $condition);
    }, function ($logger, $condition) {
        $logger->log('default', $condition);
    });

$this->assertSame([['triggered', 2]], $logger->values);
```

And the same for all other tests here.

---

And eventually, they can even become single line arrow functions:

```php
$logger = (new ConditionableLogger())->when(
    2,
    fn ($logger, $condition) => $logger->log('triggered', $condition);
    fn ($logger, $condition) => $logger->log('default', $condition);
);
```

Reply:
Yeah, I didn't love it but I wanted some assurance that the callbacks weren't triggered at all. I actually think the "throw an exception" approach works great, here, too.

Reply:
> I wanted some assurance that the callbacks weren't triggered at all

Calling the logger from within the callback, and then running an assertion that those values weren't logged (which I did above), gives you that assurance.

> actually think the "throw an exception" approach works great, here, too.

I actually prefer the logger implementation, because it keeps the tests more uniform. And it's also easier to read without having to refer back to the object to understand what those on/off methods do.